### PR TITLE
Grab textstyles from document instead of artboards

### PR DESCRIPTION
**Pull request by robbedec**

### CODE REVIEW

File: lib/mappers/adobexd/textstyle.js [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 
   return {
     token: tokenize('textStyle'),
-    id: camelcase(name.split('/').shift()),
-    weight: getFontWeight(rangedStyles.fontFamily),
-    size: rangedStyles.fontSize,
-    family: rangedStyles.fontFamily,
-    borderBottom: rangedStyles.underline ? true : null,
-    kerning: Number(rangedStyles.charSpacing.toFixed(0)) !== 0 ? parseFloat(rangedStyles.charSpacing.toFixed(1)) : null,
+    id: camelcase(document.name.split('/').shift()),
```
Comment (thibmaek) [inline lib/mappers/adobexd/textstyle.js:9]: Do Sketch/Figma id's for text styles named `Heading1/Align/Left` also get truncated to only `heading1`?

File: lib/adobexd.js [hunk 2]
```diff
@@ -23,6 +23,13 @@ async function getArtboardFiles(outputDir) {
   return artboardObjects;
 }
 
+async function getTextStyles(outputDir) {
+  const source = path.join(outputDir, 'resources', 'graphics', 'graphicContent.agc');
```
Comment (thibmaek) [inline lib/adobexd.js:27]: are document assets (text styles, components) always located here?

File: lib/mappers/adobexd/textstyle.js (hunk 1 above)
Comment (robbedec) [reply lib/mappers/adobexd/textstyle.js:9]: Yes

File: lib/adobexd.js [hunk 3]
```diff
@@ -23,6 +23,13 @@ async function getArtboardFiles(outputDir) {
...
+  const { documentLibrary } = JSON.parse(await readFile(source)).resources.meta.ux;
+
+  return documentLibrary.elements;
```
Comment (thibmaek) [inline lib/adobexd.js]: To be sure only to include text styles I would add an extra filter here:

```suggestion
  return Array.isArray(documentLibrary.elements) ? documentLibrary.elements.filter(e => e.type.startsWith('application/vnd.adobe.element.characterstyle')) : undefined;
```

**Review by thibmaek — CHANGES_REQUESTED** [review]
You still need to return the `fonts` property from the parser (see Sketch and Figma). All fonts can be collected from textstyles' `postscriptName` property.

Be sure to return a Set (array with unique elements) and not an Array for fonts

File: lib/adobexd.js (hunk 2 above)
Comment (robbedec) [reply lib/adobexd.js:27]: I suppose, that file is a collection of every resource in the XD file (artboards, documentLibrary...)

File: lib/mappers/adobexd/textstyle.js (hunk 1 above)
Comment (thibmaek) [reply lib/mappers/adobexd/textstyle.js:9]: Always? Or if the --ignore-text-style-paths flag is passed?

File: lib/adobexd.js (hunk 3 above)
Comment (robbedec) [reply lib/adobexd.js]: How should it behave if there are no textstyles defined in the file? I was thinking of returning an empty array so it doesn't crash the flow of the program.

Comment (thibmaek) [reply lib/adobexd.js]: that's fine

File: lib/adobexd.js [hunk 4]
```diff
@@ -23,6 +23,19 @@ async function getArtboardFiles(outputDir) {
   return artboardObjects;
 }
 
+async function getTextStyles(outputDir) {
+  const source = path.join(outputDir, 'resources', 'graphics', 'graphicContent.agc');
+  const { documentLibrary } = JSON.parse(await readFile(source)).resources.meta.ux;
+
+  return Array.isArray(documentLibrary.elements) ? documentLibrary.elements.filter(e => e.type.startsWith('application/vnd.adobe.element.characterstyle')) : undefined;
```
Comment (thibmaek) [inline lib/adobexd.js]: so as per your suggestion, return an empty array here

**Review by thibmaek — APPROVED** [review]

**Pull request merged**